lib/features/campaigns/helper/picture_gallery_view.dart

```diff
@@ -0,0 +1,321 @@
+            ),
+            itemBuilder: (BuildContext context) => <PopupMenuEntry<PhotoGalleryAction>>[
+              PopupMenuItem<PhotoGalleryAction>(
+                value: PhotoGalleryAction.replacePhotoWithCamera,
+                child: Row(
+                  children: [
+                    Icon(Icons.add_a_photo_outlined),
+                    SizedBox(width: 12),
+                    Text(t.campaigns.poster.photoEditActions.acquireNewPhoto),
+                  ],
+                ),
+                onTap: () => _doGalleryAction(PhotoGalleryAction.replacePhotoWithCamera),
+              ),
+              PopupMenuItem<PhotoGalleryAction>(
+                value: PhotoGalleryAction.replacePhotoWithImage,
+                child: Row(
+                  children: [
+                    Icon(Icons.image_outlined),
+                    SizedBox(width: 12),
+                    Text(t.campaigns.poster.photoEditActions.uploadFile),
+                  ],
+                ),
+                onTap: () => _doGalleryAction(PhotoGalleryAction.replacePhotoWithImage),
+              ),
+            ],
+          ),
+        ),
+      ],
+    );
+  }
+
+  void _doGalleryAction(PhotoGalleryAction result) async {
+    setCurrentImageIndex(int index) {
+      _currentImageIndex = index;
+      _imageSliderController.jumpToPage(index);
+    }
+
+    switch (result) {
+      case PhotoGalleryAction.downloadPhoto:
+        widget.downloadImage(_currentImageIndex);
+      case PhotoGalleryAction.deletePhoto:
+        if (await _showConfirmationDialog(context, PictureViewConfirmationDialogOptions.delete)) {
+          widget.removeImage(_currentImageIndex);
+          var currentImages = widget.getAllImages();
+          if (currentImages.isEmpty) {
+            _closeWindow();
+            return;
+          }
+          setCurrentImageIndex(min(_currentImageIndex, widget.getAllImages().length - 1));
+        }
+      case PhotoGalleryAction.replacePhoto:
+        throw UnimplementedError();
+      case PhotoGalleryAction.replacePhotoWithCamera:
+        if (await _showConfirmationDialog(context, PictureViewConfirmationDialogOptions.replace)) {
+          var result = await widget.replaceImageWithCamera(_currentImageIndex);
+          if (result) setCurrentImageIndex(0);
+        }
+      case PhotoGalleryAction.replacePhotoWithImage:
+        if (await _showConfirmationDialog(context, PictureViewConfirmationDialogOptions.replace)) {
+          var result = await widget.replaceImageWithDevice(_currentImageIndex);
+          if (result) setCurrentImageIndex(0);
+        }
+    }
+    setState(() {});
+  }
+
+  void _closeWindow() {
+    Navigator.maybePop(context);
+  }
+
+  Future<bool> _showConfirmationDialog(BuildContext context, PictureViewConfirmationDialogOptions options) async {
+    var title = switch (options) {
+      PictureViewConfirmationDialogOptions.replace =>
+        t.campaigns.poster.photoEditActions.confirmation_dialog.title_replace,
```

Comment:
Extreme bike-shedding, but if you want to, you could define a `tRoot` variable or so with a value of `t.campaigns.poster.photoEditActions` that's always interpolated so that it's easier to spot the difference but ... it's also ok not to ;-)